R-package/tests/testthat/test_parameters.R

```diff
@@ -101,14 +101,83 @@ test_that("training should warn if you use 'dart' boosting, specified with 'boos
         , num_threads = .LGB_MAX_THREADS
     )
     params[[boosting_param]] <- "dart"
+
+    # expect warning
+    # if early_stopping_rounds is specified
     expect_warning({
       result <- lightgbm(
         data = train$data
         , label = train$label
         , params = params
         , nrounds = 5L
         , verbose = -1L
+        , early_stopping_rounds = 1
       )
     }, regexp = "Early stopping is not available in 'dart' mode")
+
+    # expect no warning
+    # if early_stopping_rounds is not
+    # specified see:
+    # https://github.com/microsoft/LightGBM/issues/6612
+    expect_no_warning({
+      result <- lightgbm(
+        data = train$data
+        , label = train$label
+        , params = params
+        , nrounds = 5L
+        , verbose = -1L,
+        , early_stopping_rounds = NULL
+      )
+    }
+    )
   }
 })
+
+test_that(
+  desc = "lgb.cv() should only warn if you use 'dart' boosting, specified with 'boosting' or aliases",
+  code = {
+    for (boosting_param in .PARAMETER_ALIASES()[["boosting"]]) {
+      params <- list(
+        num_leaves = 5L
+        , learning_rate = 0.05
```

Comment:
```suggestion
```

Customizing `learning_rate` is not necessary to test this behavior. Please remove this. And please remove it in the other `lightgbm()` test you modified. I know it was there already (not added by your PR), but seeing it here made me realize that it's unnecessary.